clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/ObjectStorageGateway.java

```diff
@@ -630,6 +630,12 @@ public CreateBucketResponseType createBucket(final CreateBucketType request) thr
         throw new InvalidBucketNameException(request.getBucket());
       }
 
+      if (Principals.isFakeIdentify(canonicalId)) {
```

Comment:
This looks like an authorization check that should be handled by OsgAuthorizationHandler.getInstance().operationAllowed() invocation a few lines below. Why is there a separate check only in the bucket creation path?